File: app/src/main/java/org/hackillinois/androidapp2019/view/IndoorMapsFragment.java
package org.hackillinois.androidapp2019.view;

import android.os.Bundle;
import android.support.design.widget.TabLayout;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;

import org.hackillinois.androidapp2019.R;

public class IndoorMapsFragment extends Fragment {
    private static final int DCL = 0;
    private static final int ECEB = 1;
    private static final int SIEBEL = 2;

    private ImageView map;

    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_indoor_maps, container, false);

        TabLayout tabs = view.findViewById(R.id.indoor_tabs);
        tabs.addOnTabSelectedListener(new TabListener());

        map = view.findViewById(R.id.map);
        setImage(DCL);

        return view;
    }


    private class TabListener implements TabLayout.OnTabSelectedListener {
        public void onTabSelected(TabLayout.Tab tab) {
            int index = tab.getPosition();
            setImage(index);
        }

        public void onTabUnselected(TabLayout.Tab tab) {
        }

        public void onTabReselected(TabLayout.Tab tab) {
        }
    }

    private void setImage(int index) {
        switch (index) {
            case DCL:
                map.setImageResource(R.drawable.dcl_indoor);
                break;
            case ECEB:
                map.setImageResource(R.drawable.eceb_indoor);
                break;
            case SIEBEL:
                map.setImageResource(R.drawable.siebel_indoor);
                break;
        }
    }
}

File: app/src/main/java/org/hackillinois/android/view/maps/IndoorMapsFragment.kt
package org.hackillinois.android.view.maps

import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

import com.davemorrissey.labs.subscaleview.ImageSource
import com.davemorrissey.labs.subscaleview.SubsamplingScaleImageView
import kotlinx.android.synthetic.main.fragment_indoor_maps.view.*

import org.hackillinois.android.R

class IndoorMapsFragment : Fragment() {

    private var map: SubsamplingScaleImageView? = null
    private var index = 0

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val view = inflater.inflate(R.layout.fragment_indoor_maps, container, false)

        map = view.map
        map?.setMinimumTileDpi(MINIMUM_TILE_DPI)
        setImage(index)

        return view
    }

    fun onTabSelected(index: Int) {
        setImage(index)
    }

    private fun setImage(index: Int) {
        this.index = index
        map?.setImage(ImageSource.resource(imageResources[index]))
    }

    companion object {
        // TODO: get new indoor maps
        private val imageResources = intArrayOf(R.drawable.dcl_indoor, R.drawable.eceb_indoor, R.drawable.siebel_indoor, R.drawable.dcl_indoor)
        private val MINIMUM_TILE_DPI = 160
        fun newInstance() = IndoorMapsFragment()
    }
}
